File: benchmarks/string_manipulation.js
// String manipulation benchmark

// Create a large string
const size = 500000;
let text = 'a'.repeat(size);

// Start measuring time
const startTime = Date.now();

// String operations
let result = '';

// String replacement
text = text.replace(/a/g, 'b');

// String splitting and joining
const parts = text.split('b', 1000);
result = parts.join('-');

// String template literals
for (let i = 0; i < 1000; i++) {
  result = `${result.substring(0, 100)}${i}`;
}

const endTime = Date.now();
const duration = endTime - startTime;

console.log(`String size: ${size}`);
console.log(`Result length: ${result.length}`);
console.log(`Duration: ${duration}ms`);

File: dart_benchmarks/string_manipulation.dart
// String manipulation benchmark for Dart
import 'dart:core';

void main() {
  final size = 500000;
  final stopwatch = Stopwatch()..start();
  
  // Create a large string
  String text = 'a' * size;
  
  // Initialize result
  String result = '';
  
  // String replacement (equivalent to .replace(/a/g, 'b') in JS)
  text = text.replaceAll('a', 'b');
  
  // Split and join
  final parts = text.split('').take(100).toList();
  result = parts.join('-');
  
  // String operations in a loop
  for (int i = 0; i < 1000; i++) {
    result = result.substring(0, result.length > 10 ? 10 : result.length);
    result += i.toString();
  }
  
  // Stop timer
  stopwatch.stop();
  final duration = stopwatch.elapsedMilliseconds;
  
  // Output in same format as JS version
  print('String size: $size');
  print('Result length: ${result.length}');
  print('Duration: ${duration}ms');
}